pyfr/solvers/base/system.py

```diff
@@ -73,6 +73,9 @@ def __init__(self, backend, rallocs, mesh, initsoln, nregs, cfg):
         # Observed input/output bank numbers
         self._rhs_uin_fout = set()
 
+        # Track if RHS has been called
+        self.rhs_has_been_called = False
```

Comment:
This is fragile; make it the responsibility of the stepper to call rhs() with a named keyword argument saying what it the score is.

Reply:
As in keep track of how many RHS calls or just a True/False?

Reply:
A boolean argument to rhs controlling what happens is probably the most robust.